lib/vector-buffers/src/variants/disk_v2/common.rs

```diff
@@ -39,10 +44,57 @@ pub const MAX_FILE_ID: u16 = u16::MAX;
 #[cfg(test)]
 pub const MAX_FILE_ID: u16 = 6;
 
+// The alignment used by the record serializer.
+const SERIALIZER_ALIGNMENT: usize = 16;
+const MAX_ALIGNABLE_AMOUNT: usize = usize::MAX - SERIALIZER_ALIGNMENT;
```

Comment:
I have a bit of a doubt about these types (and `align16` below). If these are all on-disk sizes, and we allow >4GB buffers, should they be `u64` instead of `usize`?

Reply:
Frankly, yeah, I'm so annoyed by having to switch between `usize` and `u64`, but...

Our usage of `rkyv`, and its overaligned vector type, `AlignedVec`, means we end up overaligning anything we serialize via `rkyv` to 16 bytes. In order to calculate the correct amount to subtract from buffer limits, we use `align16` exclusively for getting the overaligned size of both the record header itself (everything we wrap around the encoded event payload) and the ledger state, which we also use `rkyv` to handle (de)serialization for.

Since `std::mem::size_of<T>` gives us back a `usize`, like most methods that return the length of a data structure, etc, it just made sense to use `usize` with `align16`. 

Practically speaking, our size limits -- both for the buffer overall and for a given data file -- work on `u64`, so we can theoretically make them bigger than 4GB but right now we're hard-coded down to 128MB for the maximum data file size and `u64::MAX` for the buffer size, save for a user overriding that with `buffer.max_size`. Records, however, cannot be larger than `usize` just from the perspective of: we encode events to a `Vec<T>`, which can only describe its length in `usize`, and so we can't actually serialize a single record larger than `usize`.

This is all a long way of saying that we specifically try to deal with `usize` only where absolutely necessary.